ebs-nvme-mapping.sh

```
@@ -1,10 +1,21 @@
 #!/bin/bash
+# Inspired by https://github.com/oogali/ebs-automatic-nvme-mapping
+# Thanks Oogali!
+# Tested on Ubuntu 16.04
+# To be used with the below udev rule (/etc/udev/rules.d/999-aws-ebs-nvme.rules)
+# SUBSYSTEM=="block", KERNEL=="nvme[0-9]*n[0-9]*", ATTRS{model}=="Amazon Elastic Block Store", PROGRAM+="/usr/local/sbin/ebs-nvme-mapping.sh /dev/%k" SYMLINK+="%c"
 
-PATH="${PATH}:/usr/sbin"
-
-for blkdev in $( nvme list | awk '/^\/dev/ { print $1 }' ) ; do
-  mapping=$(nvme id-ctrl --raw-binary "${blkdev}" | cut -c3073-3104 | tr -s ' ' | sed 's/ $//g')
-  if [[ "${mapping}" == /dev/* ]]; then
-    ( test -b "${blkdev}" && test -L "${mapping}" ) || ln -s "${blkdev}" "${mapping}"
-  fi
-done
+if [[ -x /usr/sbin/nvme ]] && [[ -b ${1} ]]; then
```

If your test conditions are not matched, this script will silently fail (with a return value of zero).

A better option would be to invert the test conditions, as to be loud and verbose that we can't find the `nvme` program and that a required argument is missing.